xinference/core/chat_interface.py

```diff
@@ -61,61 +75,67 @@ def build(self) -> "gr.Blocks":
 
     def build_chat_interface(
         self,
-        model,
-        model_info,
     ) -> "gr.Blocks":
-        model_name = model_info["model_name"]
-        model_format = model_info["model_format"]
-        model_size_in_billions = model_info["model_size_in_billions"]
-        quantization = model_info["quantization"]
-
         def flatten(matrix: List[List[str]]) -> List[str]:
             flat_list = []
             for row in matrix:
                 flat_list += row
             return flat_list
 
-        def to_chat(lst: List[str]) -> List[Dict[str, str]]:
+        def to_chat(lst: List[str]) -> List[ChatCompletionMessage]:
             res = []
             for i in range(len(lst)):
                 role = "assistant" if i % 2 == 1 else "user"
-                res.append(
-                    {
-                        "role": role,
-                        "content": lst[i],
-                    }
-                )
+                res.append(ChatCompletionMessage(role=role, content=lst[i]))
             return res
 
         def generate_wrapper(
             message: str,
             history: List[List[str]],
-            max_response_length: int,
+            max_tokens: int,
             temperature: float,
-        ) -> str:
-            output = model.chat(
+        ) -> Generator:
+            from ..client import RESTfulClient
+
+            client = RESTfulClient(self.endpoint)
+            model = client.get_model(self.model_uid)
+            assert isinstance(model, RESTfulChatModelHandle)
+
+            response_content = ""
+            for chunk in model.chat(
                 prompt=message,
                 chat_history=to_chat(flatten(history)),
                 generate_config={
-                    "max_tokens": int(max_response_length),
+                    "max_tokens": int(max_tokens),
                     "temperature": temperature,
-                    "stream": False,
+                    "stream": True,
                 },
-            )
-            return output["choices"][0]["message"]["content"]
+            ):
+                assert isinstance(chunk, dict)
+                delta = chunk["choices"][0]["delta"]
+                if "content" not in delta:
+                    continue
+                else:
+                    response_content += delta["content"]
+                    yield response_content
+
+            yield response_content
 
         return gr.ChatInterface(
             fn=generate_wrapper,
-            # TODO: Change min/max based on model context size
             additional_inputs=[
                 gr.Slider(
-                    minimum=1, maximum=2048, value=1024, step=1, label="Max Tokens"
+                    minimum=1,
+                    maximum=self.context_length,
+                    value=128,
```

Comment:
128 is too small for chat.

Reply:
This is applicable to the 'generate' function, not the 'chat' function. In this scenario, generating a large number of tokens at once may not be advisable as the generated content needs frequent updating.